apps/form-builder/services/apiService.tsx

```diff
@@ -0,0 +1,293 @@
+import axios from 'axios'
+import {
+  IForm,
+  IFormApplicantType,
+  IFormBuilder,
+  IGroup,
+  IInput,
+  ILanguage,
+  IStep,
+  ItemType,
+} from '../types/interfaces'
+
+const BASEURL = 'https://profun.island.is/umsoknarkerfi/api'
+
+export async function getForm(id: unknown) {
+  try {
+    const response = await axios.get(`${BASEURL}/Forms/${id}`)
+    console.log('getForm response: ', response.data)
+    return response.data
+  } catch (error) {
+    console.error(error)
+    throw error
+  }
+}
+
+export async function updateForm(
+  form: IForm,
+  steps: IStep[],
+  groups: IGroup[],
+  inputs: IInput[],
+) {
+  const updatedForm: IForm = {
+    ...form,
+    stepsList: steps.map((s, i) => {
+      return {
+        ...s,
+        displayOrder: i,
+      }
+    }),
+    groupsList: groups.map((g, i) => {
+      return {
+        ...g,
+        displayOrder: i,
+      }
+    }),
+    inputsList: inputs.map((i, index) => {
+      return {
+        ...i,
+        displayOrder: index,
+      }
+    }),
+  }
+
+  try {
+    const response = await axios.put(
+      `${BASEURL}/Forms/${form.id}`,
+      updatedForm,
+      {
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      },
+    )
+
+    console.log('Update complete')
+    return response
+  } catch (error) {
+    console.error('Error in updateNavbar:', error)
+    throw error
+  }
+}
+
+export async function updateItem(type: string, data: IStep | IGroup | IInput) {
+  const url = `${BASEURL}/${type}s/${data.id}`
+
+  try {
+    const response = await axios.put(url, data, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    })
+
+    console.log('response', response)
+    console.log(`saving ${type}`, data)
```

Comment:
Are all of the console.log statements in this file useful? Our infra team prefers structured JSON logs.